### PR TITLE
Add unit conversion support to detailed forecast values

### PR DESCRIPTION
**Pull request by lymanepp**
Added unit conversion support to detailed forecast values. Also changed the key type of the details dictionary to Detail instead of string.

### CODE REVIEW

File: pynws/forecast.py [hunk 1]
```diff
@@ -71,15 +103,13 @@ def last_update(self) -> datetime:
     @staticmethod
     def _find_detail_for_time(
         when, time_values: tuple[datetime, datetime, Any], units: str | None
-    ) -> tuple[Any, str | None]:
+    ) -> Value | None:
         for start_time, end_time, value in time_values:
             if start_time <= when < end_time:
-                return value, units
-        return None, None
+                return Value(value, units)
```
Comment (lymanepp) [inline pynws/forecast.py]: The value/unit tuple has been replaced with an object that can do unit conversions.

File: pynws/forecast.py [hunk 2]
```diff
@@ -88,12 +118,14 @@ def get_details_for_time(
         when = when.astimezone(timezone.utc)
         details = {}
         for detail, (time_values, units) in self.details.items():
-            details[detail] = self._find_detail_for_time(when, time_values, units)
+            details[Detail(detail)] = self._find_detail_for_time(
```
Comment (lymanepp) [inline pynws/forecast.py:122]: Convert detail string to Detail enum in result dictionaries.

File: tests/test_nws.py [hunk 3]
```diff
@@ -79,9 +80,13 @@ async def test_nws_detailed_forecast(aiohttp_client, loop, mock_urls):
...
-    assert details[Detail.TEMPERATURE] == (18.88888888888889, "degC")
-    assert details[Detail.RELATIVE_HUMIDITY] == (97.0, "percent")
-    assert details[Detail.WIND_SPEED] == (12.964, "km_h-1")
+    for key in details.keys():
+        assert isinstance(key, Detail)
+    temperature = details[Detail.TEMPERATURE]
+    assert isinstance(temperature, Value)
+    assert temperature.value == 18.88888888888889
+    assert temperature.unit == Unit.CELSIUS
+    assert temperature.value_as_unit(Unit.FAHRENHEIT) == 66
```
Comment (lymanepp) [inline tests/test_nws.py:89]: Example of using QuantitativeValue class' unit conversion

**Comment by lymanepp** [comment]
Maybe it would be better to add an option to configure metric vs imperial units. If configured, it would automatically convert values, otherwise it would return the NWS native values. It would be configured on the Nws object--which would support the SimpleForecast and Observation classes in the future.

**Comment by lymanepp** [comment]
I haven't used the SimpleNWS layer and, in fact, I hadn't even looked at it. I found that you already have unit conversion in there to force all observation units to be metric. Since the NWS already returns values in World Meteorological Organization units (wmoUnit), I think the only conversion that will actually be made is converting wind speeds from m/s to km/h.

Note: Home Assistant also uses m/s as the standard unit for wind speeds, so perhaps that conversion is not necessary.

At any rate, I'm going to close this PR and give it some more thought.

**Comment by MatthewFlamm** [comment]
Yeah I'm not sure that we should provide user configurable units.  That is probably best done in the end application.  Otherwise, it is a Pandora's box of adding units.

The unit conversions for observations was implemented as NWS servers have, in the past, temporarily switched units for observations.  I think it makes sense for this library to try and enforce consistent units when possible.

But I'm open to all thoughts still.

**Pull request closed without merging**